index.js

```diff
@@ -0,0 +1,649 @@
+    if (!isInputInTx(txb.tx, utxo.txid, utxo.index) && !(utxo.assets && utxo.assets.length)) {
+      console.log('current amount ' + utxo.value + ' needed ' + cost)
+      console.log('utxo.txid', utxo.txid)
+      console.log('utxo.index', utxo.index)
+      txb.addInput(utxo.txid, utxo.index)
+      if (txb.tx.ins.length === 1) { // encode asset
+        console.log(txb.tx.ins[0].script)
+        assetId = self._encodeAssetId(
+          args.reissueable,
+          utxo.txid,
+          utxo.index,
+          utxo.scriptPubKey.hex,
+          args.divisibility,
+          args.aggregationPolicy)
+      }
+      console.log('math: ' + current.toNumber() + ' ' + utxo.value)
+      current = current.add(utxo.value)
+      if (args.flags && args.flags.injectPreviousOutput) {
+        var chunks = bitcoinjs.script.decompile(new Buffer(utxo.scriptPubKey.hex, 'hex'))
+        txb.tx.ins[txb.tx.ins.length - 1].script = bitcoinjs.script.compile(chunks)
+      }
+      console.log('current amount: ' + current + ' projected cost: ' + cost + ' are were there yet: ' + (current.comparedTo(cost) >= 0))
+    } else {
+      console.log('skipping utxo for input, asset found in utxo: ' + utxo.txid + ':' + utxo.index)
+    }
+    return current.comparedTo(cost) >= 0
+  })
+  console.log('hasEnoughEquity: ' + hasEnoughEquity)
+  if (!hasEnoughEquity) {
+    return {success: false}
+  }
+
+  change = current - cost
+  console.log('finished adding inputs to tx')
+  console.log('change ' + change)
+  return {success: true, txb: txb, change: change, assetId: assetId, totalInputs: { amount: current }}
+}
+
+ColoredCoinsBuilder.prototype._getIssuanceCost = function (args) {
+  var self = this
+  var fee = args.fee || self.defaultFee
+  var totalCost = fee
+  console.log('_getTotalIssuenceCost: fee =', fee)
+  if (args.transfer && args.transfer.length) {
+    args.transfer.forEach(function (to) {
+      totalCost += self.mindustvalue
+    })
+  }
+
+  // TODO: calculate multisig only if actually needed
+  if (args.rules || args.metadata) {
+    totalCost += self.writemultisig ? self.mindustvaluemultisig : 0
+  }
+
+  // change
+  totalCost += self.mindustvalue
+
+  console.log('_getTotalIssuenceCost: totalCost =', totalCost)
+  return totalCost
+}
+
+ColoredCoinsBuilder.prototype._encodeAssetId = function (reissueable, txid, nvout, hex, divisibility, aggregationPolicy) {
+  var opts = {
+    ccdata: [{
+      type: 'issuance',
+      lockStatus: !reissueable,
+      divisibility: divisibility,
+      aggregationPolicy: aggregationPolicy
+    }],
+    vin: [{
+      txid: txid,
+      vout: nvout,
+      previousOutput: {
+        hex: hex
+      }
+    }]
+  }
+
+  if (!reissueable) {
+    console.log('sending assetIdEncoder locked, first input = ' + txid + ':' + nvout)
+  } else {
+    console.log('sending assetIdEncoder unlocked, first input previousOutput = ', opts.vin[0].previousOutput)
+  }
+
+  console.log('encoding asset is locked: ' + !reissueable)
```

Comment:
I suggest to replace `console.log()` with `debug()` by using npm's debug (logs can be turned on\off by giving a specific value to `DEBUG` environment variable.